#111 Enabled adding property to empty file.

amriteya opened the pull request:
This bug fix allows Butterfly's AddProperty TU to add property to an empty file. This logic sends back an empty EOL character preventing AddProperty from looking for an EOL character of a line for an empty file.

amriteya commented:
@fabiocarvalho777 
Fabio, let me know if these changes conform to your expectations.
I tried updating the existing EOL helper function rather than introducing a completely new condition for empty files.
Any suggestions for improvement are welcomed.

fabiocarvalho777 commented:
@amrish7 the idea is good, but I believe it would be better to change another method. The method you changed (`String getEndEol(String)`) does not have any idea of a file being modified. It focus only on analyzing a String and returning a value based on that analysis. So we can't consider the context of a file on that particular method.

I believe a better candidate for change would be `String findEol(File)`, because it knows it is dealing with a file, and its contract already takes that into consideration (it states `If the specified file has no EOL characters null will be returned`. I believe a good solution would be to add there a check `if file.length() == 0`, and then `return null` if true. This fix would not require changing any contract.

However, that fix would not be complete, because it would result in a new line being added in the empty file before adding the new property. To avoid that, I believe a check `if file.length() == 0` should also be added at `AddProperty.addProperty`, and only if true the EOL character should be added.

What do you think? Does that make sense?

amriteya commented:
@fabiocarvalho777 I second the suggestion. It does not make sense to have empty character being returned from `String getEndEol(String)` method.
However, I don't understand the reason why we need to have a `file.length()` check in `String findEol(String)` method if we are going to have that check in `AddProperty.addProperty` method.

Example:
```java
if (fileToBeChanged.length() != 0) {
   FileUtils.fileAppend(fileToBeChanged.getAbsolutePath(), EolHelper.findEolDefaultToOs(fileToBeChanged));
}
```

amriteya commented:
Ahh! I did not think of that. Yes, it makes complete sense. Let me update my PR.

amriteya commented:
@fabiocarvalho777 Updated the PR.

Recent changes:
- Added logic for empty file in `AddProperty.addProperty` and `EolHelper.findEol`.
- Updated documentation for `EolHelper.findEol`.
- Added unit tes for empty file condition in`EolHelper.findEol`.

Let me know if this suffices or if there is still scope for improvement.
